vm/src/asm.rs

```diff
@@ -498,6 +498,9 @@ pub struct Assembler
 
     /// Current section
     section: Section,
+
+    // Code Metadata
+    meta: HashMap<u64, OpMeta>,
```

Comment:
This seems useful and has been requested by another user.

Only comment I have would be, could you rename `meta` to `op_meta`, because eventually there's going to be program-level metadata as well.